google/cloud/bigquery/client.py

```diff
@@ -342,9 +355,15 @@ def list_datasets(
             # and converting it into a string here.
             extra_params["filter"] = filter
         path = "/projects/%s/datasets" % (project,)
+        span_attributes = {"path": path, "page_token": page_token}
+
+        with create_span(
+            name="BigQuery.listDatasets", attributes=span_attributes, client=self
+        ):
+            api_request = functools.partial(self._call_api, retry, timeout=timeout)
```

Comment:
Same idea applies here. You'll need to define an `api_request` function rather than use functools.partial if you want the span to be created when the actual API request happens.

Reply:
Acknowledged in a separate PR as per our conversation. Draft can be found here:
https://github.com/googleapis/python-bigquery/pull/228